### ISSUE TITLE
Enable support for "Apple IIc RamExpress II+" clock!

### ISSUE THREAD

**Issue by jewettg** [a2stuff/a2d#802]
**Is your feature request related to a problem? Please describe.**
I do not see the clock appear in the upper-right of my menu bar.
I am using: [Apple IIc RamExpress II+](https://www.a2heaven.com/webshop/index.php?rt=product/product&product_id=144) which has clock capabilities.

**Describe the solution you'd like**
Standard "clock.system" driver does not seem to work, unaware of any drivers for this card.

**Describe alternatives you've considered**
(none).  Can set date and time and it will be kept during power-on.

**Additional context**
Still doing some research, will update appropriately.

**Comment by inexorabletash** [comment]
Clock driver is from https://github.com/a2stuff/prodos-drivers - can you file an issue over there?

**Comment by inexorabletash** [comment]
Actually... per https://www.a2heaven.com/webshop/index.php?rt=product/product&product_id=144 the clock is DCLOCK compatible and CLOCK.SYSTEM includes DCLOCK support. 

You can try copying DCLOCK.SYSTEM from the linked disk image there into your boot directly and adjusting the directory order so that it comes before CLOCK.SYSTEM (or replace CLOCK.SYSTEM) and it should attempt to install a ProDOS clock driver.

Also... it's a quirk of CLOCK.SYSTEM that it can only detect the type of clock you have if the date/time is already set and valid. If you haven't set the date/time using the DCLOCK utility disk, CLOCK.SYSTEM can't verify that a clock is present and the driver won't be installed. Have you used the DCLOCK utility disk to verify the date/time are set?

**Comment by jewettg** [comment]
I am re-opening this issue.  I do not feel it was resolved, **and I am happy to help debug/troubleshoot.**
I have tried putting "dclock.system" from the prodos drivers disk into the root-level of my boot.  Made no difference.  I removed the "clock.system" file and left just "dclock.system", and still no clock.

On top of that, when I copy the "dclock.setup" program from the prodos drivers diskette, it tells me that no dclock is found.  I mean it flashes ever so briefly, but that is what it says.  I expect, if it found it, it would let me set the date and time.

I am going to open a ticket with "info@a2heaven.com" to see if they have any suggestions?!

**Comment by inexorabletash** [comment]
> Also... it's a quirk of CLOCK.SYSTEM that it can only detect the type of clock you have if the date/time is already set and valid. If you haven't set the date/time using the DCLOCK utility disk, CLOCK.SYSTEM can't verify that a clock is present and the driver won't be installed. Have you used the DCLOCK utility disk to verify the date/time are set?

This applies to the DCLOCK.SYSTEM from the prodos-drivers disk as well. 

You need to start with the DCLOCK disk from https://www.a2heaven.com/webshop/index.php?rt=product/product&product_id=144 - don't worry about A2D until you've got a working setup without it.

Your goal should be to boot into ProDOS using the DCLOCK disk and ensure the dedicated clock driver is installed and the time and date is set. Verify by saving a file to disk and using `CAT` to verify the time stamp is set correctly. Until you are able to do that, don't confuse things by trying other disks.

If you have that working - i.e. you have ProDOS booted, and have verified that you have a working clock driver resident in memory - then try starting A2D by launching DESKTOP.SYSTEM. If not, then there's a problem with A2D and we can re-open this. But until you've done the preliminary work, it's premature to assume there's a problem with A2D.

**Comment by jewettg** [comment]
I can not remember his name over at a2heaven!  But we can now officially CLOSE this.

Apparently the ProDOS drivers disk does not function or have the proper drivers.  The disk image from his website is the one to use, but I could not get it to work.  He told me that I would need to rename it from *.dsk to *.po, and then I would be able to mount it on my //c and boot it.

Sure enough.. it booted to the AE tools, I was able to set the clock, and patch the RAMDISK installation of A2Desktop.. and it works!  I now have a clock in the upper-right corner!  Amazing.   

A2Desktop is amazing!  
![IMG_2968](https://github.com/user-attachments/assets/f239bf6c-01d5-483b-a758-fb643e0dba0d)

**Comment by jewettg** [comment]
Sorry for the duplicate issue -- I could not figure out how to re-open.  :-(

**Comment by inexorabletash** [comment]
Try this disk image for DCLOCK utilities:

https://mirrors.apple2.org.za/ftp.apple.asimov.net/unsorted/DClock_Y2K_MGFix.dsk

**Comment by inexorabletash** [comment]
Glad you have it working! https://github.com/a2stuff/prodos-drivers/tree/main/clocks/dclock notes that DCLOCK support in the drivers collection is untested... must be a glitch that snuck in somewhere. I'll see about fixing, but tricky without the hardware.

**Comment by mgcaret** [comment]
I attempted to reproduce this issue using a IIc with a RAM Express II+ installed. Battery was dead so I replaced it with a new one and the onboard time was invalid. As expected, the clock drivers do not install with bad time. 

I set the time with the Dclock utilities and cold-booted the latest A2D beta. Time display was present. 

I then cold-booted ProDOS, ran the DCLOCK.SYSTEM from the lates prodos-drivers disk, and then started A2D.  Time display was present. 

Could not find any issue with the CLOCK.SYSTEM on the A2D disk, nor the DCLOCK.SYSTEM from prodos-drivers.
